### ISSUE TITLE
IDE0010 should not complain about missing default case?

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#65326]
### Discussed in https://github.com/dotnet/roslyn/discussions/65322

<div type='discussions-op-text'>

<sup>Originally posted by **rafael-f** November  9, 2022</sup>
Shouldn't IDE0010 be separated into two rules? So that we can disable one and keep the other?

This is what IDE0010 checks for: (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0010)

-An enum switch statement that's missing cases for one or more enum members.
-A switch statement with a missing default case.

Problem is, I see no reason for it to complain about missing default case when all the other enum members are already in the switch.

I think the rule should either be separated, so people who want default in all their switches can use it, or the rule should not require default case if all the other enum members are in the switch already...

The probem with the current behavior is that if I want to have the rule activate in case someone adds something to the enum but forgets to add to the switch, I cannot disable the rule, and if I leave the rule on I cannot reach the default case in unit tests and this leaves the case uncovered on the coverage report.</div>

**Comment by CyrusNajmabadi** [comment]
I'll take this.

**Comment by gp-jorge** [comment]
Hi @CyrusNajmabadi, did you have an update on this? Is there any approach to fix it? It would be nice to report that the default is missing but if there is a default, it should not be complaining about the missing cases, right?

**Comment by CyrusNajmabadi** [comment]
@gp-jorge can you give a screenshot of what you mean?

**Comment by CyrusNajmabadi** [comment]
Closing due to lack of response.

**Comment by glen-84** [comment]
@CyrusNajmabadi

@gp-jorge is referring to this:

```cs
switch (value) // IDE0010: Populate switch
{
    case Example.Value1:
        break;
    // case Example.Value2:
    //     break;
    default:
        break;
}
```

Though it may be better to handle the missing values explicitly?

Regardless, the original issue remains:

```cs
switch (value) // IDE0010: Populate switch
{
    case Example.Value1:
        break;
    case Example.Value2:
        break;
    // default:
    //     break;
}
```

This is acceptable for some people, at least in application (non-library) code.

As it stands, you cannot receive warnings for missing values without also receiving warnings for a missing default case.
